src/VirtoCommerce.Platform.Web/Controllers/ExternalSignInController.cs

```diff
@@ -48,6 +48,12 @@ public ActionResult SignIn(string authenticationType)
         [AllowAnonymous]
         public async Task<ActionResult> SignInCallback(string returnUrl)
         {
+            var decodedReturnUrl = System.Web.HttpUtility.UrlDecode(returnUrl, System.Text.Encoding.Default);
+            if (!Uri.TryCreate(decodedReturnUrl, UriKind.Absolute, out var parsedReturnUrl))
```

Comment:
Couldn't built-in `IUrlHelper.IsLocalUrl` [link](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.iurlhelper.islocalurl?view=aspnetcore-3.1) be used here?

Reply:
Yep.